### ISSUE TITLE
Lacking functional / regression tests for catalysis.

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#2004]
As noted in https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1998#issuecomment-670811489_
there was a bug recently introduced that broke all heterogeneous catalysis simulations, but was not detected by any of the automated testing. Presumably this means we don't have any catalysis examples in the test suite. We should add one.

**Comment by JacksonBurns** [comment]
This is a need-to-have more than a nice-to-have - adding this to the long term projects board.

**Comment by JacksonBurns** [comment]
This is in progress [#2415](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2514)

**Comment by sevyharris** [comment]
We'll need to update the rmgpy.tools.regression tool to be able to handle surface mechanisms. I'll create a separate issue/PR for that

**Comment by hwpang** [comment]
I have this PR back in Jun to add regression test for RMS surface reactor, but it was blocked by the fact that we weren't able to compare surface model because surface species and gas/liquid phase species are saved separately (see my initial comment here: https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2470#issuecomment-1598966294).

**Comment by sevyharris** [comment]
https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2563 needs to be merged before https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2514 in order to update the rmgpy.tools.regression tool for surface mechanisms

**Comment by JacksonBurns** [comment]
I have rebased the PR that will close this issue, #2514. @hwpang can you review?
